File: backend/src/main/java/com/example/workout_app/models/logs/RoutineLog.java
package com.example.workout_app.models.logs;

import java.time.LocalDateTime;
import java.util.List;

import com.example.workout_app.models.Account;
import com.example.workout_app.models.defaults.Routine;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table
@Data
@SequenceGenerator(name = "routine_log_seq", sequenceName = "routine_log_seq", allocationSize = 1)
public class RoutineLog {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    private Account account;

    @OneToMany(mappedBy = "routineLogs")
    private List<WorkoutLog> workoutLogs;

    @ManyToOne
    @JoinColumn(name = "routine_id")
    private Routine refRoutine;

    private String name;
    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;
}

File: CSharpBackend/Models/Logs/RoutineLog.cs
using System.ComponentModel.DataAnnotations;

namespace FitnessBackend.Models;
public class RoutineLog
{
    [Key]
    public long Id { get; set; }
    public required string Name {get; set;}
    public DateTime startDateTime { get; set; }
    public DateTime endDateTime { get; set; }

    // Foreign key
    public required string UserId { get; set; }
    public long RefRoutineId { get; set; }
    public List<long> WorkoutLogIds { get; set; } = new List<long>();

    // Navigation property
    public required User User { get; set; }
    public Routine? RefRoutine { get; set; }
    public List<WorkoutLog> WorkoutLogs { get; set; } = new List<WorkoutLog>();
}